roles/translator/src/downstream_sv1/downstream.rs

```diff
@@ -46,23 +52,24 @@ pub struct Downstream {
 
 impl Downstream {
     /// Instantiate a new `Downstream`.
-    pub async fn new(
+    #[allow(clippy::while_let_loop)]
+    pub async fn new_downstream(
         stream: TcpStream,
         tx_sv1_submit: Sender<(v1::client_to_server::Submit<'static>, Vec<u8>)>,
         rx_sv1_notify: Receiver<server_to_client::Notify<'static>>,
         extranonce1: Vec<u8>,
         last_notify: Option<server_to_client::Notify<'static>>,
         target: Vec<u8>,
         extranonce2_len: usize,
-    ) -> ProxyResult<'static, Arc<Mutex<Self>>> {
+    ) {
         let stream = std::sync::Arc::new(stream);
 
         // Reads and writes from Downstream SV1 Mining Device Client
         let (socket_reader, socket_writer) = (stream.clone(), stream);
         let (tx_outgoing, receiver_outgoing) = bounded(10);
 
         let socket_writer_clone = socket_writer.clone();
-        let _socket_writer_set_difficulty_clone = socket_writer.clone();
+        // let _socket_writer_set_difficulty_clone = socket_writer.clone();
```

Comment:
I'm assuming you removed this because it was never used? Might as well just delete it.